tests/testthat/test-carehomes-multistrain.R

```diff
@@ -1476,32 +1850,148 @@ test_that("G_D empty when p_G_D = 0", {
+
+  mod <- carehomes$new(p, 0, np, seed = 1L)
+
+  initial <- carehomes_initial(mod$info(), np, p)
+  mod$set_state(initial$state, initial$step)
+  end <- sircovid_date("2020-05-01") / p$dt
+  steps <- seq(initial$step, end, by = 1 / p$dt)
+  set.seed(1)
+  y <- mod$transform_variables(
+    drop(mod$simulate(steps)))
+
+  expect_true(all(y$S[, , , 85] == 0))
+  expect_true(all(y$R[, 2:4, , , 85] == 0))
+  expect_false(all(y$R[, 1, , , 85] == 0))
+})
+
+
+test_that("Can only move from R3 and R4 to S", {
+  np <- 3L
+  p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                            initial_I = 0,
+                            strain_transmission = c(1, 1),
+                            strain_seed_rate = c(10, 0),
+                            waning_rate = 1 / 5,
+                            model_pcr_and_serology = 0,
+                            strain_seed_date =
+                              sircovid_date(c("2020-02-07", "2020-02-08")),
+                            model_super_infection = 1)
+  ## Prevent anyone leaving S
+  p$rel_susceptibility[] <- 0
+
+  mod <- carehomes$new(p, 0, np, seed = 1L)
+  info <- mod$info()
+
+  initial <- carehomes_initial(info, np, p)
+  y0 <- initial$state
+  ## Empty R1 and R2
+  y0[info$index$R][1:38] <- 0
+  ## Fill R3 and R4
+  y0[info$index$R][39:76] <- 1e3
+
+  mod$set_state(y0)
+
+  end <- sircovid_date("2020-05-01") / p$dt
+  steps <- seq(initial$step, end, by = 1 / p$dt)
+  set.seed(1)
+  y <- mod$transform_variables(
+    drop(mod$simulate(steps)))
+
+  diff_S <- y$S[-4, , , 2] - y$S[-4, , , 1]
+  diff_R <- y$R[-4, 3:4, , , 1] - y$R[-4, 3:4, , , 2]
+  diff_R <- apply(diff_R, c(1, 3), sum)
+  expect_equal(diff_R, diff_S)
+  expect_true(all(y$E[-4, , , , , ] == 0))
+})
+
+
+test_that("Everyone in R3 and R4 when no waning and transmission high", {
+  np <- 1L
+  p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                            strain_transmission = c(1, 1),
+                            strain_seed_rate = c(10, 0),
+                            strain_seed_date =
+                              sircovid_date(c("2020-02-07", "2020-02-08")),
+                            model_super_infection = 1,
+                            model_pcr_and_serology = 0)
+  p$strain_transmission[] <- 1e8
+
+  mod <- carehomes$new(p, 0, np, seed = 1L)
+
+  initial <- carehomes_initial(mod$info(), np, p)
+  mod$set_state(initial$state, initial$step)
+  end <- sircovid_date("2021-05-01") / p$dt
+  steps <- seq(initial$step, end, by = 1 / p$dt)
+  set.seed(1)
+  y <- mod$transform_variables(
+    drop(mod$simulate(steps)))
 
-  # Strain 1
-  expect_false(all(y[index_G_D_strain_1, , ] == 0))
-  expect_false(all(y[index_G_D_strain_1, , ] == 0))
-  # Strain 2
-  expect_true(all(y[index_G_D_strain_2, , ] == 0))
-  expect_true(all(y[index_G_D_strain_2, , ] == 0))
+  expect_true(all(y$R[, 1:2, , 450] == 0))
+  expect_false(all(y$R[, 3:4, , 450] == 0))
```

Comment:
I was going to suggest you could do better than this and check that the sum of individuals in R[3:4] at last time step is equal to the total population (p$N_tot), or not far off. But when I checked this locally, turns out there is almost noone in R[4], and that remains true even when I set strain_seed_rate to something crazy high. So I am not sure the seeding works properly... This needs checking

Reply:
Found the same, which is why the values are set so high